src/lib/storage/table.cpp

```diff
@@ -71,6 +72,16 @@ Chunk& Table::get_chunk(ChunkID chunk_id) { return _chunks.at(chunk_id); }
 
 const Chunk& Table::get_chunk(ChunkID chunk_id) const { return _chunks.at(chunk_id); }
 
-void Table::compress_chunk(ChunkID chunk_id) { throw std::runtime_error("Implement Table::compress_chunk"); }
+void Table::compress_chunk(ChunkID chunk_id) {
+  auto new_chunk = Chunk();
```

Comment:
Add thread safety:
`std::mutex _access_mutex;` as member variable in table.hpp
`std::lock_guard lock(_access_mutex);` - in the beginning of compress_chunk, get_chunk (const), append, add_column